save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/topbar/TopBarLinks.kt

```diff
@@ -50,13 +52,40 @@ data class TopBarLink(
     val isExternalLink: Boolean = false,
 )
 
+private fun ChildrenBuilder.demoDropdownEntry(
+    name: String,
+    href: String,
+    navigate: NavigateFunction,
+    deactivateDropdown: () -> Unit,
+) {
+    dropdownEntry(null, name, window.location.href.contains(href)) { attrs ->
+        attrs.onClick = {
+            deactivateDropdown()
+            navigate(to = href)
+        }
+    }
+}
+
 /**
  * Displays the static links that do not depend on the url.
  */
 @Suppress("MAGIC_NUMBER", "LongMethod", "TOO_LONG_FUNCTION")
 private fun topBarLinks() = FC<TopBarLinksProps> { props ->
     val navigate = useNavigate()
     var isDemoDropdownActive by useState(false)
+    val deactivateDropdown = { isDemoDropdownActive = false }
+    var demos by useState(listOf<DemoDto>())
```

Comment:
You should not want to use `useState` for lists with `by` initialization as setting state for lists usually means passing lambda to `StateSetter`, which as I know is impossible with delegation

Reply:
thanks for a note. Here `demos` is set a fully without manipulation with list, but will rewrite to `pair` to avoid possible problems in future